pkg/client/unversioned/services.go

```diff
@@ -98,3 +99,16 @@ func (c *services) Watch(label labels.Selector, field fields.Selector, resourceV
 		FieldsSelectorParam(field).
 		Watch()
 }
+
+func (c *services) ProxyGet(name, path string, params map[string]string) ([]byte, error) {
```

Comment:
It's probably a tradeoff.  I would suggest doing the following - define a
new interface in the client package "Response" (or generically named) like:

type Response interface {
  Raw() ([]byte, error)
  Stream() (io.Reader, error)
}

and then return that from this method.  Clients would therefore choose
whether to use Raw() or Stream().  In your method you can return Request
because it satisfies this interface.


On Tue, Aug 25, 2015 at 9:37 AM, Jerzy Szczepkowski <
notifications@github.com> wrote:

> In pkg/client/unversioned/services.go
> <https://github.com/kubernetes/kubernetes/pull/13134#discussion_r37859393>
> :
>
> > @@ -98,3 +99,16 @@ func (c *services) Watch(label labels.Selector, field fields.Selector, resourceV
> >  		FieldsSelectorParam(field).
> >  		Watch()
> >  }
> > +
> > +func (c *services) ProxyGet(name, path string, params map[string]string) ([]byte, error) {
>
> So, how can I make the result more generic? If I execute Do() (even w/o
> Raw) the whole response will be put into memory. Do you think Stream() will
> be better?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/13134/files#r37859393>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift